File: lib_c/array.c
// The Lumen library file lib_lumen/array.lm, ported by scripts/port_examples.py; edit the Lumen original, not this file.

long array_index_of(long a, long x) {
    long i = 0;
    while (i < strlen(a)) {
        if (a[i] == x) {
            return i;
        }
        i = i + 1;
    }
    return -1;
}

bool array_contains(long a, long x) {
    return array_index_of(a, x) >= 0;
}

File: langs/lib_pascal/array.pas
// The Lumen library file langs/lib_lumen/array.lm, ported by scripts/port_examples.py; edit the Lumen original, not this file.

function array_index_of(a: integer; x: integer): integer;
var i: integer;
begin
    i := 0;
    while i < length(a) do begin
        if a[i] = x then begin
            exit(i);
        end;
        i := i + 1;
    end;
    array_index_of := -1;
end;

function array_contains(a: integer; x: integer): boolean;
begin
    array_contains := array_index_of(a, x) >= 0;
end;
